src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -575,7 +575,10 @@ BoundBlock runAnalysis(BoundBlock block, DiagnosticBag blockDiagnostics)
                     // rather we go directly to LowerBodyOrInitializer, which skips over flow analysis (which is in CompileMethod)
                     // (the same thing - calling ControlFlowPass.Analyze in the lowering - is done for lambdas)
                     // It's a bit of code duplication, but refactoring would make things worse.
-                    var endIsReachable = ControlFlowPass.Analyze(localSymbol.DeclaringCompilation, localSymbol, block, blockDiagnostics);
+                    // However, we don't need to report diagnostics here. They will be reported when analyzing the parent method.
+                    var ignored = DiagnosticBag.GetInstance();
```

Comment:
This is just getting rid of that `Label is not referenced` warning, right?

Reply:
This is actually just preventing duplicates from showing up. The reason we are getting rid of those warnings is because we're analyzing the whole method for control flow now, instead of just a scoped segment, so we actually do see that the label is being referenced, just in a way that's illegal (so we provide an error elsewhere)